src/data-workspace/inputs/option-set.js

```diff
@@ -11,6 +18,15 @@ import {
 import styles from './inputs.module.css'
 import { InputPropTypes } from './utils.js'
 
+const MULTI_TEXT_SEPERATOR = ','
+
+// This is used to preserve the order of the optionSet for the selected options
+const createSortByOptionsOrder = (options) => (a, b) => {
+    const aIndex = options.findIndex((option) => option.code === a)
+    const bIndex = options.findIndex((option) => option.code === b)
+    return aIndex - bIndex
+}
```

Comment:
Pretty sure this is OK because in the end it's just to parse a few selected values. But this function is super expensive.

Reply:
Yeah, you're right. But we currently don't have any searching functionality in options, so hopefully the optionSets shouldn't be that huge. I did consider converting to a map-object, but decided it's probably not worth it. 
Also, this only happens when actually changing the selection. So there would need to be a lot of options for this to actually matter.

Reply:
We could of course find both indexes in one pass, by doing something like this.

```js
const createSortByOptionsOrderOnePass = (options) => (a, b) => {
    let aIndex, bIndex
    for (let i = 0; i < options.length; i++) {
        if (aIndex && bIndex) {
            break
        }
        const option = options[i]
        if (option.code === a) {
            aIndex = i
        }
        if (option.code === b) {
            bIndex = i
        }
    }
    return aIndex - bIndex
}
```